plugins/flatpak/gs-flatpak.c

```diff
@@ -442,6 +442,11 @@ gs_flatpak_mark_apps_from_usb_remote (GsFlatpak *self,
 				 app_collection_id, ref_collection_id);
 			continue;
 		}
+
+		// TODO: to the USB category (possibly not in here), add top
```

Comment:
Yeah, plugins cannot change the UI directly. So this needs to be like you had said: add it to the category page UI file, then show it only when appropriate.

Reply:
Will do